### PR TITLE
Fix adding to ACL in SL0 and appropriate test case

### PR DESCRIPTION
**Pull request by finikorg**

### CODE REVIEW

File: tbtadm/controller.cpp [hunk 1]
```diff
@@ -746,12 +746,17 @@ void tbtadm::Controller::acl()
...
+    switch (m_sl)
     {
-        m_out << "Adding to ACL on SL2 must be done together with device "
-                 "approval\n";
-        return;
+        case 2:
+            m_out << "Adding to ACL on SL2 must be done together with device "
+                    "approval\n";
+            return;
+        case 0:
```
Comment (ybernat) [inline tbtadm/controller.cpp]: OK, if we want this, it should be similar to line 560 above (or even refactored into a function called from both locations). SL3 is also a thing.

**Review by ybernat — CHANGES_REQUESTED** [review]
"Resolves #44"?

File: tbtadm/controller.cpp (hunk 1 above)
Comment (finikorg) [reply tbtadm/controller.cpp]: I think we need to preserve error messages which are different for every security level. Or do you mean we need to refactor also line 560 the same way? I can also include `default` case for other levels.

Comment (ybernat) [reply tbtadm/controller.cpp]: This must capture only SL1. Any other SL isn't supported here, each one for its own reason.

**Review by ybernat — CHANGES_REQUESTED** [review]
If SL0 must be handle, so is other non-SL1 modes

**Comment by finikorg** [comment]
Added enum and more checks. The test case for the bug is 1st commit, you can verify that it fails without the fix.

**Review by ybernat — APPROVED** [review]
Nicely done. Thanks!

**Pull request merged**